From ed36a063becc81651dddd0807fb43ec759209d8c Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Tue, 26 Nov 2019 19:35:04 +0100
Subject: [PATCH 1025/1239] mvebu: spl: add boot from nand option

After switching from per board kwbimage.cfg to more generic solution done
in ca1a4c863232ddc6bc2463569dedfefe992ff418
("mvebu: select boot device at SoC level") the ability of choosing NAND as
boot device was missing. Add such options in common makefile which will
allow to generate required spl format. (tools/kwbimage.c is already capable
of proceeding nand spl image).

Change-Id: Id55ccf0c4e0b70f6becfe646e0d4ed867c452cab
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Reviewed-on: https://sj1git1.cavium.com/19343
Tested-by: Kostya Porotchkin <kostap@marvell.com>
Reviewed-by: Kostya Porotchkin <kostap@marvell.com>
---
 arch/arm/mach-mvebu/Kconfig  | 3 +++
 arch/arm/mach-mvebu/Makefile | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/arch/arm/mach-mvebu/Kconfig b/arch/arm/mach-mvebu/Kconfig
index 1b650888f7..09d871ee27 100644
--- a/arch/arm/mach-mvebu/Kconfig
+++ b/arch/arm/mach-mvebu/Kconfig
@@ -248,6 +248,9 @@ choice
 config MVEBU_SPL_BOOT_DEVICE_SPI
 	bool "SPI NOR flash"
 
+config MVEBU_SPL_BOOT_DEVICE_NAND
+	bool "NAND flash"
+
 config MVEBU_SPL_BOOT_DEVICE_MMC
 	bool "SDIO/MMC card"
 	select SPL_LIBDISK_SUPPORT
diff --git a/arch/arm/mach-mvebu/Makefile b/arch/arm/mach-mvebu/Makefile
index 30ea111b5b..1bf147440b 100644
--- a/arch/arm/mach-mvebu/Makefile
+++ b/arch/arm/mach-mvebu/Makefile
@@ -33,6 +33,9 @@ KWB_REPLACE += BOOT_FROM
 ifneq ($(CONFIG_MVEBU_SPL_BOOT_DEVICE_SPI),)
 	KWB_CFG_BOOT_FROM=spi
 endif
+ifneq ($(CONFIG_MVEBU_SPL_BOOT_DEVICE_NAND),)
+	KWB_CFG_BOOT_FROM=nand
+endif
 ifneq ($(CONFIG_MVEBU_SPL_BOOT_DEVICE_MMC),)
 	KWB_CFG_BOOT_FROM=sdio
 endif
-- 
2.29.0

